weave

```diff
@@ -572,49 +613,42 @@ case "$COMMAND" in
         echo weave script $SCRIPT_VERSION
         ask_version $CONTAINER_NAME $IMAGE
         ask_version $DNS_CONTAINER_NAME $DNS_IMAGE
-        if ! TOOLS_IMAGE_ID=$(docker inspect --format='{{.Id}}' $TOOLS_IMAGE 2>/dev/null) ; then
-            echo "Unable to find $TOOLS_IMAGE image." >&2
+        if ! EXEC_IMAGE_ID=$(docker inspect --format='{{.Id}}' $EXEC_IMAGE 2>/dev/null) ; then
+            echo "Unable to find $EXEC_IMAGE image." >&2
         else
-            TOOLS_VERSION=$(docker images --no-trunc | grep $TOOLS_IMAGE_ID | grep -v latest | tr -s ' ' | cut -d ' ' -f 2 | tr "\n" ' ')
-            if [ -n "$TOOLS_VERSION" ] ; then
-                echo "weave tools $TOOLS_VERSION"
+            EXEC_VERSION=$(docker images --no-trunc | grep $EXEC_IMAGE_ID | grep -v latest | tr -s ' ' | cut -d ' ' -f 2 | tr "\n" ' ')
+            if [ -n "$EXEC_VERSION" ] ; then
+                echo "weave exec $EXEC_VERSION"
             else
-                echo "weave tools (unreleased version)"
+                echo "weave exec (unreleased version)"
             fi
         fi
         ;;
     run)
         [ $# -gt 0 ] || usage
-        if [ "$1" = "--with-dns" ] ; then
-            shift 1
-            if [ \( "$DOCKER_VERSION_MAJOR" -lt 1 \) -o \
-                 \( "$DOCKER_VERSION_MAJOR" -eq 1 -a \
-                    "$DOCKER_VERSION_MINOR" -lt 2 \) ] ; then
-                echo "ERROR: The '--with-dns' option requires Docker 1.2.0 or later; you are running $DOCKER_VERSION" >&2
-                exit 1
-            fi
-            docker_bridge_ip
-            DNS_ARG="--dns $DOCKER_BRIDGE_IP"
-            DNS_SEARCH_ARG="--dns-search=."
-            for arg in "$@"; do
-                case $arg in
-                    --dns-search=*)
-                        DNS_SEARCH_ARG=""
-                        ;;
-                    *)
-                        ;;
-                esac;
-            done
+        if [ "$LOCAL" = "--local" ]; then
+            dns_args "$@"
+        else
+            DNS_ARGS=$(exec_remote dns-args "$@")
+            [ "$1" = "--with-dns" ] && shift 1
         fi
+        validate_cidr $1
```

Comment:
Is there a reason you moved this? Or is it just a merge gone wrong? Everywhere else we first set CIDR and then invoke validate_cidr.